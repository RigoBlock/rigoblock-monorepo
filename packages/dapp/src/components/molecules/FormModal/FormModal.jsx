import './FormModal.scss'
import PropTypes from 'prop-types'
import React from 'react'
import WrapperWithDivider from '../../molecules/WrapperWithDivider'

const divider = () => <div className="modal-divider" />

const FormModal = ({ Header, Form }) => {
  return (
    <div className="form-modal">
      <WrapperWithDivider Divider={divider}>
        <div className="form-header">
          <Header />
        </div>
        <div className="form-content">
          <Form />
        </div>
      </WrapperWithDivider>
    </div>
  )
}

FormModal.propTypes = {
  Form: PropTypes.func.isRequired,
  Header: PropTypes.func.isRequired
}

export default FormModal
