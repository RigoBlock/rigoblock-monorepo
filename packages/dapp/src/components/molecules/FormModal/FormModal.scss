@import "../../_settings/variables.scss";

.form-modal {
  max-width: 9.25 * $dist-l;
  background: $secondary-bg-color;

  .form-header, .form-content {
    padding: $dist-s;
  }

  .modal-divider {
    background: $dark-grey;
    height: 1px;
  }
}

