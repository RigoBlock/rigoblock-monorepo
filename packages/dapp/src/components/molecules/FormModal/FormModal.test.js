import FormModal from './FormModal.jsx'
import React from 'react'
import toJson from 'enzyme-to-json'

const header = () => <div>Header Component</div>
const form = () => <div>Header Component</div>
const props = {
  Header: header,
  Form: form
}

describe('FormModal component', () => {
  it('renders correctly', () => {
    expect(toJson(createComponentWithProps(FormModal, props))).toMatchSnapshot()
  })
})
