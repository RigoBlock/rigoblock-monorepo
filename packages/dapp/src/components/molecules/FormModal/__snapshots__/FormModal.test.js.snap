// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormModal component renders correctly 1`] = `undefined`;
